contracts/src/StakeTable.sol

```diff
@@ -129,11 +153,45 @@ contract StakeTable is AbstractStakeTable {
             && BN254.BaseField.unwrap(a.y1) == BN254.BaseField.unwrap(b.y1);
     }
 
-    /// TODO handle this logic more appropriately when epochs are re-introduced
-    /// @dev Fetches the current epoch from the light client contract.
-    /// @return current epoch (computed from the current block)
-    function currentEpoch() public pure returns (uint64) {
-        return 0;
+    /// @dev Fetches the last hotshot block number from the light client contract to calculate the
+    /// epoch.
+    /// @return current epoch (computed from the last known hotshot block number)
+    function currentEpoch() public view returns (uint64) {
+        // get the last hotshot block number from the light client contract
+        (, uint64 lastHotshotBlockNumber,) = lightClient.finalizedState();
+
+        // calculate the epoch from the last hotshot block number
+        uint64 epoch = lastHotshotBlockNumber / hotShotBlocksPerEpoch;
+
+        return epoch;
+    }
+
+    /// @notice get the next available epoch based and queue size
+    /// @param firstAvailableEpoch The first available epoch (for registration or exit)
+    /// @param numPending The number of pending exits or registrations
+    /// @return epoch The next available epoch
+    /// @return queueSize The size of the queue in that epoch
+    /// TODO handle overflow when max uint64 is reached
+    function nextAvailableEpoch(uint64 firstAvailableEpoch, uint64 numPending)
+        external
+        view
+        override
+        returns (uint64, uint64)
+    {
+        uint64 epoch;
+        uint64 queueSize;
+
+        if (firstAvailableEpoch < currentEpoch() + 1) {
```

Comment:
Add a comment as to why we don't need to worry about overflow?

Reply:
i added it as a todo in line 174, i want to start a discussion around this, will do so now

Reply:
so they don't handle overflow in the offchain code, so i think a revert if it overflows can work? though it's a check every time this is invoked. what do you think?

Reply:
added a test to show that it panics/reverts when currentEpoch() == max(uint64) https://github.com/EspressoSystems/espresso-sequencer/commit/b215c7f80cad672413ed0e29c7b47d85aaac94c4